packages/redux-dynostore-core/src/detachableReducer.js

```diff
@@ -6,12 +6,10 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-export const DETACH_TYPE = 'DYNOSTORE/DETACH_REDUCER'
+const DETACH_TYPE = '@@DYNOSTORE/DETACH_REDUCER'
 
-const detachable = (identifier) => (reducer) => (state, action) => {
-  return (action.identifier === identifier && action.type === DETACH_TYPE)
-    ? null
-    : reducer(state, action)
-}
+export const detach = identifier => ({ type: DETACH_TYPE, identifier })
 
-export default detachable
+export const detachable = identifier => reducer => (state, action) => {
+  return action.type !== DETACH_TYPE || action.identifier !== identifier ? reducer(state, action) : undefined
```

Comment:
The custom `combineReducers` implementation.  Many of the common warnings in redux come from there, and they're there to help identify issues and help beginners get over hurdles.  As the combining and merging happens in within this library (and therefore impervious to errors 😉), we don't benefit from them, particularly when because of the weird things we are trying to achieve.

The `filteredReducer` was a work around for the fact the the `combineReducers` creates a bunch of warnings as reducers get added/replaced, so a nice benefit of making our own is that we don't have to wrap everything in yet another higher-order reducer (which has performance overheads).  We still wrap the initial static reducer as we cannot control what the users of this library want to give use, although we could export out implementation(s) and document the warning and tell them to use ours instead, but I like not being so prescriptive on that at the moment.

Reply:
It may also be worth setting up a scenario to see what happens in a reducer is detached when a deeper nested reducer has been added, e.g.

```js
store.addReducers({ test: reducer1, 'test.nested': reducer2 })
store.detachReducers(['test'])
```

To be honest, I'm not sure what it will do or what I expect to happen. Thoughts?